app/gui2/src/components/DocumentationPanel.vue

```diff
@@ -0,0 +1,266 @@
+import { default as DocsSynopsis } from '@/components/documentation/DocsSynopsis.vue'
+import { default as DocsTags } from '@/components/documentation/DocsTags.vue'
+import { useSuggestionDbStore, type SuggestionDb } from '@/stores/suggestionDatabase'
+import type { SuggestionEntry, SuggestionId } from '@/stores/suggestionDatabase/entry'
+import { SuggestionKind, entryQn } from '@/stores/suggestionDatabase/entry'
+import type { Doc } from '@/util/docParser'
+import { type Opt } from '@/util/opt'
+import type { SuggestionEntryArgument } from 'shared/languageServerTypes/suggestions'
+import { computed } from 'vue'
+
+const props = defineProps<{ selectedEntry: Opt<SuggestionEntry> }>()
+
+// === Types ===
+
+/**
+ * Intermediate representation of the entries documentation.
+ */
+export type Docs =
+  | { Function: FunctionDocs }
+  | { Type: TypeDocs }
+  | { Module: ModuleDocs }
+  | { Local: LocalDocs }
+  | { Placeholder: string }
+
+export interface FunctionDocs {
+  name: string
+  arguments: SuggestionEntryArgument[]
+  sections: Sections
+}
+
+export interface TypeDocs {
+  name: string
+  arguments: SuggestionEntryArgument[]
+  sections: Sections
+  methods: FunctionDocs[]
+  constructors: FunctionDocs[]
+}
+
+export interface ModuleDocs {
+  name: string
+  sections: Sections
+  types: TypeDocs[]
+  methods: FunctionDocs[]
+}
+
+export interface LocalDocs {
+  name: string
+  sections: Sections
+}
+
+// === Sections ===
+
+/**
+ * Documentation sections split into three categories. These categories of sections are present in almost every documentation page.
+ */
+export interface Sections {
+  tags: Doc.Section.Tag[]
+  synopsis: Doc.Section[]
+  examples: Doc.Section.Marked[]
+}
+
+// Split doc sections into categories.
+function filterSections(sections: Iterable<Doc.Section>): Sections {
+  const tags = []
+  const synopsis = []
+  const examples = []
+
+  for (const section of sections) {
+    const isTag = 'Tag' in section
+    const isExample = 'Marked' in section && section.Marked.mark == 'Example'
+    if (isTag) {
+      tags.push(section.Tag)
+    } else if (isExample) {
+      examples.push(section.Marked)
+    } else {
+      synopsis.push(section)
+    }
+  }
+  return { tags, synopsis, examples }
+}
+
+// === Lookup ===
+
+function getChildren(db: SuggestionDb, id: SuggestionId, kind: SuggestionKind): Docs[] {
+  if (!id) return []
+  const children = Array.from(db.parent.reverseLookup(id)).map((id) => db.get(id))
+  return children.reduce((acc: Docs[], child: SuggestionEntry | undefined) => {
+    if (child?.kind === kind) {
+      const docs = lookupDocumentation(db, child)
+      acc.push(docs)
+    }
+    return acc
+  }, [])
+}
+
+function asFunctionDocs(docs: Docs[]): FunctionDocs[] {
+  return docs.flatMap((doc) => {
+    if ('Function' in doc) {
+      return [doc.Function]
+    } else {
+      console.error(`Unexpected docs type: ${docs}, expected Function`)
+      return []
+    }
+  })
+}
```

Comment:
It's awkward, but I didn't think of a better way. I guess I need to change the definition of the `Docs` type somehow because the current one causes too many type acrobatics in the code. I’d prefer to do that in a separate PR, though.